component-interaction/src/app/shared/user.service.ts

```diff
@@ -0,0 +1,104 @@
+import { Injectable } from "@angular/core";
+
+import { InputDateEnum } from "./input-date.enum";
+import { UserModel } from "./user.model";
+
+@Injectable({
+	providedIn: 'root',
+})
+export class UserService {
+	private selectedUsersIds: number[] = [];
+	private dateType = InputDateEnum;
+	private users: UserModel[] = [
+		{
+			id: 7,
+			name: "Catalin",
+			contact: "example@email.xyz",
+			dateOfBirth: new Date(2005, 1, 1)
+		},
+
+		{
+			id: 6,
+			name: "Daniel",
+			contact: "example@email.xyz",
+			dateOfBirth: new Date(2000, 1, 1)
+		},
+
+		{
+			id: 1,
+			name: "Popesco",
+			contact: "example@email.xyz",
+			dateOfBirth: new Date(2012, 1, 1)
+		},
+
+		{
+			id: 2,
+			name: "Ana",
+			dateOfBirth: new Date(1990, 1, 1)
+		}
+	];
+
+	public getUsers(): UserModel[] {
+		return this.users;
+	}
+
+	public deleteUser(id: number): void {
+		let index: number = 0;
+		for (let i = 0; i < this.users.length; i++) {
+			if (this.users[i].id === id) {
+				index = i;
+			}
+		}
```

Comment:
see another good way
const foundUser = this.users.find((user: UserModel) => user.id === id)
const index = this.users.indexOf(foundUser);